plugins/backup/backroll/src/main/java/org/apache/cloudstack/backup/backroll/BackrollClient.java

```diff
@@ -0,0 +1,628 @@
+                throw new CloudRuntimeException("Failed to retrieve backups status for this VM via Backroll");
+            }
+        } catch (final IOException | InterruptedException e) {
+            LOG.error(String.format("Failed to check backups status due to: %s", e.getMessage()));
+        }
+        return null;
+    }
+
+    public List<BackrollVmBackup> getAllBackupsfromVirtualMachine(String vmId) {
+        LOG.info(String.format("Trying to retrieve all backups for vm %s", vmId));
+
+        List<BackrollVmBackup> backups = new ArrayList<BackrollVmBackup>();
+
+        try {
+            HttpResponse backupsResponse = get(String.format("/virtualmachines/%s/backups", vmId));
+
+            if(isResponseOk(backupsResponse)) {
+                HttpEntity body = backupsResponse.getEntity();
+                String bodyStr = EntityUtils.toString(body);
+
+                BackrollTaskRequestResponse backupTaskRequestResponse = new ObjectMapper().readValue(bodyStr, BackrollTaskRequestResponse.class);
+                String urlToRequest = backupTaskRequestResponse.location.replace("/api/v1", "");
+                LOG.debug(urlToRequest);
+                EntityUtils.consumeQuietly(backupsResponse.getEntity());
+
+                bodyStr = waitForGetRequestResponse(urlToRequest);
+
+                VirtualMachineBackupsResponse response = new ObjectMapper().readValue(bodyStr, VirtualMachineBackupsResponse.class);
+                EntityUtils.consumeQuietly(backupsResponse.getEntity());
+
+                if(response.state.equals("SUCCESS")) {
+                    if(response.info.archives.size() > 0) {
+                        for (BackupInfos infos : response.info.archives) {
+                            var dateStart = new DateTime(infos.start);
+                            backups.add(new BackrollVmBackup(infos.id, infos.name, dateStart.toDate()));
+                        }
+                    }
+                }
+            }
+
+        } catch (IOException | InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        return backups;
+    }
+
+    private HttpResponse post(final String path, final JSONObject json) throws IOException {
+        String xml = null;
+        StringEntity params = null;
+        if (json != null) {
+            LOG.debug("JSON " + json.toString());
+            params = new StringEntity(json.toString(), ContentType.APPLICATION_JSON);
+        }
+
+        String url = apiURI.toString() + path;
+        final HttpPost request = new HttpPost(url);
+
+        if (params != null) {
+            request.setEntity(params);
+        }
+
+        request.setHeader(HttpHeaders.AUTHORIZATION, "Bearer " + backrollToken);
+        request.setHeader("Content-type", "application/json");
+
+        final HttpResponse response = httpClient.execute(request);
+
+        LOG.info(String.format("Response received in POST request with body [%s] is: [%s] for URL [%s].", xml, response.toString(), url));
```

Comment:
```suggestion
        LOG.debug(String.format("Response received in POST request with body [%s] is: [%s] for URL [%s].", xml, response.toString(), url));
```
Also, use log4j2 format